homeassistant/components/braviatv/config_flow.py

```diff
@@ -0,0 +1,191 @@
+
+from .const import (  # pylint:disable=unused-import
+    ATTR_CID,
+    ATTR_MAC,
+    ATTR_MODEL,
+    CLIENTID_PREFIX,
+    CONF_IGNORED_SOURCES,
+    DOMAIN,
+    NICKNAME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def host_valid(host):
+    """Return True if hostname or IP address is valid."""
+    try:
+        if ipaddress.ip_address(host).version == (4 or 6):
+            return True
+    except ValueError:
+        disallowed = re.compile(r"[^a-zA-Z\d\-]")
+        return all(x and not disallowed.search(x) for x in host.split("."))
+
+
+class BraviaTVConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for BraviaTV integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self.braviarc = None
+        self.host = None
+        self.title = None
+        self.mac = None
+
+    async def init_device(self, pin):
+        """Initialize Bravia TV device."""
+        await self.hass.async_add_executor_job(
+            self.braviarc.connect, pin, CLIENTID_PREFIX, NICKNAME,
+        )
+
+        if not self.braviarc.is_connected():
+            raise CannotConnect()
+
+        system_info = await self.hass.async_add_executor_job(
+            self.braviarc.get_system_info
+        )
+
+        await self.async_set_unique_id(system_info[ATTR_CID].lower())
+        self._abort_if_unique_id_configured()
+
+        self.title = system_info[ATTR_MODEL]
+        self.mac = system_info[ATTR_MAC]
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Bravia TV options callback."""
+        return BraviaTVOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        self.host = user_input[CONF_HOST]
+        self.braviarc = BraviaRC(self.host)
+
+        try:
+            await self.init_device(user_input[CONF_PIN])
+        except CannotConnect:
+            _LOGGER.error("Import aborted, cannot connect to %s", self.host)
+            return self.async_abort(reason="cannot_connect")
+        except (KeyError, TypeError):
+            _LOGGER.error("Import aborted, your TV is not supported")
+            return self.async_abort(reason="unsupported_model")
+
+        user_input[CONF_MAC] = self.mac
+
+        return self.async_create_entry(title=self.title, data=user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                if not host_valid(user_input[CONF_HOST]):
+                    raise InvalidHost()
+
+                self.host = user_input[CONF_HOST]
+                self.braviarc = BraviaRC(self.host)
+
+                return await self.async_step_authorize()
+            except InvalidHost:
+                errors[CONF_HOST] = "invalid_host"
```

Comment:
Remove this raise and invert and outdent the host_valid check instead.
```suggestion
        if user_input is not None:
            if host_valid(user_input[CONF_HOST]):
                self.host = user_input[CONF_HOST]
                self.braviarc = BraviaRC(self.host)

                return await self.async_step_authorize()
            else:
                errors[CONF_HOST] = "invalid_host"
```